foxglove_bridge_base/src/test/test_client.cpp

```diff
@@ -118,6 +112,30 @@ std::future<Service> waitForService(std::shared_ptr<ClientInterface> client,
   return future;
 }
 
+std::future<Channel> waitForChannel(std::shared_ptr<ClientInterface> client,
+                                    const std::string& topicName) {
+  auto promise = std::make_shared<std::promise<Channel>>();
```

Comment:
Why shared_ptr? The std::promise example code seems to allocate a promise on the stack and still move it into a thread or lambda.

Reply:
I remember that John and I had some issues with that but don't remember exactly why. Will see what I can do once I created a new release